src/peergos/server/space/JdbcQuotas.java

```diff
@@ -123,6 +126,44 @@ public boolean hasUser(String username) {
         }
     }
 
+    public boolean allowAndRemoveToken(String token) {
```

Comment:
I still need to figure out how to handle the case where the core node reject the signup because of rate limiting. In that case we don't want to consume the token.

Reply:
Ok that's fixed now. Technically not atomic, but I'm not concerned about that.